small tweak and master (check the last few commits for the actual diff)

mvo5 opened the pull request.

AlexandreAbreu commented on client/packages.go:
```diff
@@ -161,7 +161,9 @@ func (client *Client) Find(opts *FindOptions) ([]*Snap, *ResultInfo, error) {
-	q.Set("section", opts.Section)
+	if opts.Section != "" {
+		q.Set("section", opts.Section)
+	}
```
This was actually done on purpose since the semantic of an empty snap search is to return the list of featured snaps by default.

mvo5 replied:
Is this the intended semantic for the snapd client when communicating with the snapd server? I understand this is what the store is doing (and I added a comment there because it confused me initially :)

mvo5 commented:
@AlexandreAbreu Thanks for your comment. You raise an important point. I pushed ef811bf  to make the section/featured handling explicit. Please let me know if that addresses the concerns.
